### PR TITLE
dsg-2113: applying schemas api changes

### PR DESCRIPTION
**Pull request by derekdiep42**
# Basics

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

Using the new schemas api changed implemented through this ticket: https://dol-ewf.atlassian.net/browse/DSG-2122 to search for schemas. Now loading will be fixed for schemas. Also added a test case.

## Screenshot of the affected changes

Didn't show before:
Screenshot 2023-10-19 at 5 24 51 PM

## Checklist

- [ ] I have updated the documentation accordingly
- [x] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/agency/feature/admin/src/lib/components/schemas-list/schemas-list.component.ts [hunk 1]
```diff
@@ -72,15 +72,23 @@ export class SchemasListComponent {
...
+      this._workApiRoutesService
+        .getSchemaDefinitionsList$(
+          [
+            { field: 'key', value: searchText },
+            { field: 'name', value: searchText },
+          ],
+          this.pagingRequestModel,
+        )
+        .pipe(
+          tap(x => {
```
Comment (g8shen) [inline libs/agency/feature/admin/src/lib/components/schemas-list/schemas-list.component.ts:84]: Should we have a better name then x?

**Review by g8shen — APPROVED** [review]
Small nit: otherwise LGTM!

File: libs/agency/feature/admin/src/lib/components/transaction-definitions-form/transaction-definitions-form.component.spec.ts [hunk 2]
```diff
@@ -55,6 +58,7 @@ describe('TransactionDefinitionsFormComponent', () => {
         }),
         MockProvider(TransactionDefinitionsFormService, {
           loadSchemas$: jest.fn().mockImplementation(() => of(SchemaDefinitionListSchemaMock)),
+          schemas$: schemasSubject.asObservable(),
```
Comment (kevincheng0) [inline libs/agency/feature/admin/src/lib/components/transaction-definitions-form/transaction-definitions-form.component.spec.ts]: can probably replace with `of(SchemaDefinitionListMock)`

**Review by kevincheng0 — APPROVED** [review]
nice

**Review by g8shen — APPROVED** [review]

**Pull request merged**